qutebrowser/browser/webkit/network/pac.py

```diff
@@ -0,0 +1,529 @@
+
+            args1[argtype] = arg
+            my_order.append(argtype)
+            count += 1
+
+        if len(args) == 2 * count:
+            for e, i in zip(my_order, itertools.count(0)):
+                args2[e] = args[count + i]
+        elif len(args) == count:
+            args2 = args1
+        else:
+            raise EvalProxyError("Invalid number of arguments")
+
+        if len(args1) == 0:
+            raise EvalProxyError("Nothing is passed to check the range")
+
+        return self._date_range(args1, args2, gmt)
+
+    def _time_range(self, args1, args2, gmt):
+        """Test if current time is between two time points.
+
+        Args:
+            args1, args2: Two dictionaries with optional "hour", "min" and
+            "sec".
+            gmt: Should the UTC time be used.
+        """
+        def _to_tuple(args):
+            return (args.get("hour"), args.get("min"), args.get("sec"))
+
+        ranges = (3600, 60, 1)
+
+        t = self._get_dt(gmt).time()
+        val = (t.hour(), t.minute(), t.second())
+
+        return self._in_range(_to_tuple(args1), val, _to_tuple(args2), ranges)
+
+    @_js_slot(QVariant)
+    def timeRange(self, args):
+        """Test if current time is between two time points.
+
+        If only a single value is specified (from each category: hour,
+        minute, second), the function returns a true value only at times
+        that match that specification. If both values are specified, the
+        result is true between those times, including bounds, but the bounds
+        are ordered.
+
+        Order: hour, min, sec. Next argument can't be specified without
+        previous.
+        """
+        order = ("hour", "min", "sec")
+
+        # Same as dateRange but simpler because need to autodetect less (at
+        # least seconds always follow minutes always follow hours...).
+        if isinstance(args, QJSValue):
+            args = args.toVariant()
+        args1 = {}
+        args2 = {}
+        gmt = args[-1] == "GMT"
+        if gmt:
+            args = args[:-1]
+
+        if len(args) == 1:
+            args1["hour"] = args[0]
+            args2["hour"] = args[0] + 1
+        else:
+            if len(args) % 2 != 0 or len(args) > len(order) * 2:
+                raise EvalProxyError("Invalid number of args")
+            nargs = int(len(args) / 2)
+
+            for e, i in zip(order[:nargs], itertools.count(0)):
+                args1[e] = args[i]
+                args2[e] = args[nargs + i]
+
+        if len(args1) == 0:
+            raise EvalProxyError("Nothing is passed to check the range")
+
+        return self._time_range(args1, args2, gmt)
+
+
+class PACResolver(object):
+    """Evaluate PAC script files and resolve proxies."""
+
+    @staticmethod
+    def _parse_proxy_entry(proxy_str):
+        """Parse one proxy string entry, as described in PAC specification."""
+        config = [c for c in proxy_str.split(' ') if c != ""]
```

Comment:
Same thing about explicit checks (to reiterate -- I would be happy to change how would you see fit, I'm just making sure this isn't just a hint "you can also do it like this in Python").